primap2/pm2io/_data_reading.py

```diff
@@ -191,37 +205,55 @@ def convert_long_dataframe_if(
     if add_coords_cols:
         check_overlapping_specifications_add_cols(coords_cols, add_coords_cols)
 
-    filter_data(data_long, filter_keep, filter_remove)
+    # if desired make a copy to keep input dataframe unchanged
+    if copy_df:
+        data_copy = data_long.copy(deep=True)
+    else:
+        data_copy = data_long
+
+    filter_data(data_copy, filter_keep, filter_remove)
 
     add_dimensions_from_defaults(
-        data_long, coords_defaults, additional_allowed_coords=["time"]
+        data_copy, coords_defaults, additional_allowed_coords=["time"]
     )
 
     naming_attrs = rename_columns(
-        data_long, coords_cols, add_coords_cols, coords_defaults, coords_terminologies
+        data_copy, coords_cols, add_coords_cols, coords_defaults, coords_terminologies
     )
-
     attrs.update(naming_attrs)
 
+    if isinstance(convert_str, bool):
+        if convert_str:
+            convert_str = {}
+
+    if isinstance(convert_str, dict):
+        # get data columns (just one as we have long format)
+        data_cols = ["data"]
+        # find all string values
+        str_values = find_str_values_in_data(data_copy, data_cols)
+        # create replacement dict
+        str_repl_dict = create_na_replacement_dict(str_values, convert_str)
+        replace_na_values(data_copy, data_cols, str_repl_dict)
+
```

Comment:
The function silently does nothing if I e.g. pass a string for `convert_str`. Would be better to raise a ValueError. Maybe the whole type handling can be shifted into `create_na_replacement_dict`? I could make a patch if you concur that an error should be raised.

Reply:
I agree that raising an error makes sense. I've changed it, push will come soon